Something is wrong with github CI

I get an error: 
```
##[error]        heater_regulators_test.go:163: HTR.CSVHeaders() = [HTR1A HTR1B HTR1OD HTR2A HTR2B HTR2OD HTR7A HTR7B HTR7OD HTR8A HTR8B HTR8OD WARNINGS], want [HTR1A HTR1B HTR1OD HTR2A HTR2B HTR2OD HTR7A HTR7B HTR7OD HTR8A HTR8B HTR8OD]
```
on my branch on github, but not on my machine. The error indicates that the tests were actually not run on my branch as it is but rather on how my branch would be when I update against master. This is a rather confusing and annoying behaviour.

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -20,9 +20,9 @@ jobs:
       run : sudo apt-get install -y gcc-mingw-w64
 
     - name: Check out code into the Go module directory
-       uses: actions/checkout@v1
-       with:
-         ref: ${{ github.head_ref }} 
+      uses: actions/checkout@v1
+      with:
+        ref: ${{ github.head_ref }} 
 
     - name: Get dependencies
       run: |

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -2,9 +2,6 @@ name: Go
 
 on:
   push:
-    branches: [ master ]
-  pull_request:
-    branches: [ master ]
 
 jobs:
 
@@ -23,7 +20,9 @@ jobs:
       run : sudo apt-get install -y gcc-mingw-w64
 
     - name: Check out code into the Go module directory
-      uses: actions/checkout@v2
+       uses: actions/checkout@v1
+       with:
+         ref: ${{ github.head_ref }} 
 
     - name: Get dependencies
       run: |

